Description: [droidcamsrc] notify sensor orientation change when device changed
 sensor-orientation is tied to the selected camera-device. Thus, it made
 more sense to move the notify here.
Author: Ratchanan Srirattanamet <ratchanan@ubports.com>
Forwarded: no
Last-Update: 2020-06-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/gst/droidcamsrc/gstdroidcamsrc.c
+++ b/gst/droidcamsrc/gstdroidcamsrc.c
@@ -357,6 +357,9 @@
         GST_INFO_OBJECT (src, "camera device set to %d", src->camera_device);
       }
       g_rec_mutex_unlock (&src->dev_lock);
+
+      g_object_notify (G_OBJECT (src), "sensor-orientation");
+      g_object_notify (G_OBJECT (src), "sensor-mount-angle");
       break;
 
     case PROP_MODE:
@@ -682,9 +685,6 @@
             src->dev->info->direction, src->mode, quirk, TRUE);
       }
 
-      g_object_notify (G_OBJECT (src), "sensor-orientation");
-      g_object_notify (G_OBJECT (src), "sensor-mount-angle");
-
       /* now that we have camera parameters, we can update min and max ev-compensation */
       gst_droidcamsrc_update_ev_compensation_bounds (src);
 
